stream/stream_file.c

```diff
@@ -299,12 +299,15 @@ static int open_f(stream_t *stream, const struct stream_open_args *args)
     if (strncmp(url, "fd://", 5) == 0 || is_fdclose) {
         char *begin = strstr(stream->url, "://") + 3, *end = NULL;
         p->fd = strtol(begin, &end, 0);
-        if (!end || end == begin || end[0]) {
+        if (!end || end == begin || end[0] || p->fd < 0) {
             MP_ERR(stream, "Invalid FD: %s\n", stream->url);
             return STREAM_ERROR;
         }
         if (is_fdclose)
             p->close = true;
+#ifndef _WIN32
+        assert(fcntl(p->fd, F_GETFD) != -1);
```

Comment:
Also this really should be an assert? Looks like an user runtime input error. Which should be handled with error code.

Reply:
I see this as a last resort check for something that should never happen, because fds don't just disappear.
There can also be unintended consequences if the fd number is reused before mpv tries to use it.

Reply:
> I see this as a last resort check for something that should never happen

I don't understand. `mpv fd://123` shouldn't crash. It should error out nicely.

Reply:
If you provide an invalid pointer to a library you wouldn't expect a nice error message either.
But I guess it's not directly comparable.